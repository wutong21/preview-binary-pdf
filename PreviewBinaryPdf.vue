<template>
  <iframe
    :src="fileUrl"
    :scrolling="scrolling"
    frameborder="0"
    :type="iframeType"
    :width="width"
    :style="`height:${height}`"
  ></iframe>
</template>

<script>
export default {
  name: "PreviewBinaryPdf",
  props: {
    url: { type: String, default: "" },
    width: { type: String, default: "100%" },
    height: { type: String, default: "100vh" },
    scrolling: { type: String, default: "auto" },
    type: { type: String, default: "application/pdf;charset=utf-8" },
    data: { type: ArrayBuffer, default: null },
  },
  watch: {
    data(val) {
      if (val !== null) {
        this.binaryDataToUrl(val)
      }
    },
    url(val) {
      if (val !== null) {
        this.fileUrl = this.url
      }
    }
  },
  data() {
    return {
      fileUrl: "",
      iframeType: "application/x-google-chrome-pdf"
    };
  },
  methods:{
    binaryDataToUrl(data){
      let blob = new Blob([data], {
        type: `${this.type}`
      })
      this.fileUrl = window.URL.createObjectURL(blob)
    }
  }
};
</script>
