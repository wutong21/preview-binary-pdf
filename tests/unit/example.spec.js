import { mount } from '@vue/test-utils'
import PreviewBinaryPdf from '../../PreviewBinaryPdf.vue'

describe('PreviewBinaryPdf.vue', () => {
  it('是否渲染了iframe组件', () => {
    const wrapper = mount(PreviewBinaryPdf)
    expect(wrapper.element).toMatchSnapshot()
  })
  test('is a Vue instance', () => {
    const wrapper = mount(PreviewBinaryPdf)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
