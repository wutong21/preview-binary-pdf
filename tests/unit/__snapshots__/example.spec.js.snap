// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewBinaryPdf.vue 是否渲染了iframe组件 1`] = `
<iframe
  frameborder="0"
  scrolling="auto"
  src=""
  style="height: 100vh;"
  type="application/x-google-chrome-pdf"
  width="100%"
/>
`;
